x/uibc/uics20/memo_handler.go

```diff
@@ -58,13 +58,20 @@ func (mh *MemoHandler) onRecvPacketPrepare(
 		return nil, sdkerrors.Wrap(err, "can't parse ftData.Receiver bech32 address")
 	}
 
+	var memoPayload []byte
 	if strings.EqualFold(ftData.Sender, gmp.DefaultGMPAddress) {
 		events = append(events, "axelar GMP transaction")
 		mh.isGMP = true
-		return events, nil
+		gh := gmp.NewHandler()
+		gmpMessage, err := gh.OnRecvPacket(*ctx, mh.received, mh.memo, mh.receiver)
+		if err != nil {
```

Comment:
```suggestion
		gmpMessage, err := gmp.ParseMemo(*ctx, mh.received, mh.memo, mh.receiver)
		if err != nil {
```

We don't need a handler struct. Let's simplify